pkg/volume/host_path/host_path.go

```diff
@@ -210,3 +233,24 @@ func (r *hostPathRecycler) Recycle() error {
 	}
 	return volume.ScrubPodVolumeAndWatchUntilCompletion(pod, r.host.GetKubeClient())
 }
+
+// hostPathDeleter deletes a hostPath PV from the cluster.
+// This deleter only works on a single host cluster and is for testing purposes only.
+type hostPathDeleter struct {
+	name string
+	path string
+	host volume.VolumeHost
+}
+
+func (r *hostPathDeleter) GetPath() string {
+	return r.path
+}
+
+// Delete for hostPath removes the local directory so long as it is beneath /tmp.
+// Plugins with storage APIs (GCE, AWS, etc) should use the API for deletion of the resource from the infrastructure.
+func (r *hostPathDeleter) Delete() error {
+	if !strings.HasPrefix(r.GetPath(), "/tmp") {
+		return fmt.Errorf("host_path deleter only supports /tmp/*")
+	}
+	return os.RemoveAll(r.GetPath())
```

Comment:
Maybe just regex it as `/tmp/.+`  (or test that there is some element beyond /tmp/)

You know my feeling about using HostPath as a test vehicle for PV - the more powerful it becomes the more likely it is to cause some user to nuke their system.  Can we bolster the scary warning in the comment in PersistentVolumeSource?  The scarier the better. :)